UserRelated/Main.py

```diff
@@ -0,0 +1,208 @@
+
+
+def add_difficulty_shingle(shing_set: set, args: argparse.Namespace):
+    """
+    Adds the appropriate difficulty shingle to the shingles set.
+    :param shing_set: the user preferences shingle set
+    :param args: the given command-line arguments.
+    """
+    if args.difficulty == 1:
+        shing_set.add(TrackDifficulty.EASY.value)
+    elif args.difficulty == 2:
+        shing_set.add(TrackDifficulty.INTERMEDIATE.value)
+    else:
+        shing_set.add(TrackDifficulty.DIFFICULT.value)
+
+
+def add_shape_shing(shing_set: set, args: argparse.Namespace):
+    """
+    Adds the appropriate shape shingle to the shingles set.
+    :param shing_set: the user preferences shingle set
+    :param args: the given command-line arguments.
+    """
+    shing_set.add(TrackShape.LOOP.value) if args.shape == 1 else shing_set.add(TrackShape.CURVE.value)
+
+
+# Todo: this is a code duplication of the slopes_poc branch, we should have a separate class for LSH-ing we can reuse.
+def get_min_hash(shingles: set) -> MinHash:
+    """
+    given a set of shingles, creates a MinHash object updated with those shingles.
+    :param shingles: a set of track shingles.
+    :return: a MinHash object updated with the given shingles.
+    """
+    track_min_hash = MinHash(num_perm=128)
+    for shin in shingles:
+        track_min_hash.update(str(shin).encode('utf-8'))
+    return track_min_hash
+
+
+def create_user_shingles(args: argparse.Namespace) -> set:
+    """
+    Creates a set of shingles defining the user's path preferences.
+    :param args: the given command-line arguments.
+    :return: a set of shingles as described.
+    """
+    shing = set()
+    add_interest_points_shingles(shing, args)
+    add_length_shingle(shing, args)
+    add_difficulty_shingle(shing, args)
+    add_shape_shing(shing, args)
+    return shing
+
+
+def get_osm_tracks(area_path: str) -> dict:
+    """
+    Reads a dictionary with the data collected on paths in the given search area.
+    :param area_path: path to the file containing the data relevant to the given search are.
+    :return: a dictionary with data on tracks in the given area.
+    """
+    with open(area_path, 'r') as f:
+        osm_tracks_dict = json.load(f)
+    return osm_tracks_dict['tracks']
+
+
+def in_geo_limits(args: argparse.Namespace, track_data: dict) -> bool:
+    """
+    Checks if the given track is in the geographic limits the user had given.
+    :param args: command lines arguments.
+    :param track_data: a dictionary containing a track's data.
+    :return: true if the track lies in the geographic boundaries the user had set, false otherwise.
+    """
+    return (track_data['boundaries']['north'] <= args.north_lim and
+            track_data['boundaries']['south'] >= args.south_lim and
+            track_data['boundaries']['east'] <= args.east_lim and
+            track_data['boundaries']['west'] >= args.west_lim)
+
+
+if __name__ == '__main__':
```

Comment:
i think that this can be expended to a multi-query user loop:) the USER_ID should be a counter inside main()  to handle this. but ok, maybe we're not there yet...

Reply:
It sounds good! But we are really not there yet. I add it to the project backlog and resolves the issue since it doesn't interferes with the current merge, but more of an idea for future improvement.